Source/WebGPU/WebGPU/ShaderModule.mm

```diff
@@ -111,15 +112,19 @@
 
     auto checkResult = WGSL::staticCheck(fromAPI(shaderModuleParameters->wgsl.code), std::nullopt, { maxBuffersPlusVertexBuffersForVertexStage() });
 
-    if (std::holds_alternative<WGSL::SuccessfulCheck>(checkResult) && shaderModuleParameters->hints && descriptor.hintCount) {
-        // FIXME: re-enable early compilation later on once deferred compilation is fully implemented
-        // https://bugs.webkit.org/show_bug.cgi?id=254258
-        UNUSED_PARAM(earlyCompileShaderModule);
+    if (std::holds_alternative<WGSL::SuccessfulCheck>(checkResult)) {
+        if (shaderModuleParameters->hints && descriptor.hintCount) {
+            // FIXME: re-enable early compilation later on once deferred compilation is fully implemented
+            // https://bugs.webkit.org/show_bug.cgi?id=254258
+            UNUSED_PARAM(earlyCompileShaderModule);
+        }
     } else {
-        // FIXME: remove shader library generation from MSL after compiler bringup
-        auto library = ShaderModule::createLibrary(device(), String::fromUTF8(shaderModuleParameters->wgsl.code), fromAPI(descriptor.label));
-        if (library)
-            return ShaderModule::create(WTFMove(checkResult), { }, { }, library, *this);
+        auto& failedCheck = std::get<WGSL::FailedCheck>(checkResult);
+        for (const auto& error : failedCheck.errors) {
+            // FIXME: https://bugs.webkit.org/show_bug.cgi?id=250442
+            dataLogLn(error);
```

Comment:
This is one place I'll agree that we probably shouldn't use `dataLog`, since it will actually log in production. I tried to look at how we were doing production logging in WebGPU, but it seems like we still need to work on the proper logging as per the bugzilla, and since  `CompilationMessage` only had convenient ways to print with `dataLog`, I left it as is for now.